From de4b07119fc004121fae604a9cb323cde8b07fc0 Mon Sep 17 00:00:00 2001
From: Volker Krause <vkrause@kde.org>
Date: Tue, 3 Feb 2026 17:54:18 +0100
Subject: [PATCH 1/3] Actually search for ZXing 3

Now that the version number has been bumped upstream we need to explicitly
allow for 3.0 as well.
---
 CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b375dda..14f111f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,10 +43,12 @@ find_package(Dmtx)
 set_package_properties(Dmtx PROPERTIES
     PURPOSE "Required for generation of Data Matrix barcodes."
     TYPE RECOMMENDED)
-find_package(ZXing 2.0)
-if (NOT TARGET ZXing::ZXing)
-  find_package(ZXing 1.2.0)
-endif()
+foreach(ver 3.0 2.0 1.2.0)
+    find_package(ZXing ${ver} CONFIG QUIET)
+    if (TARGET ZXing::ZXing)
+        break()
+    endif()
+endforeach()
 set_package_properties(ZXing PROPERTIES
     PURPOSE "Required for generation of PDF417 barcodes and for scanning of barcodes from live video feed."
     TYPE RECOMMENDED)
-- 
2.52.0

